### ISSUE TITLE
Content canView should check membership status and container visibility for owner of content

### ISSUE THREAD

**Issue by fae7em** [humhub/humhub#4713]
### What steps will reproduce the problem?
A member creates private content in space.
Member cancels his/her membership in that space.

### What is the expected result?
Since the content is private and this user is not a member anymore, the Content `canView` should return `false`.

### What do you get instead?
The Content `canView` just check ownership of content and return `true`.

### Additional info

| Q                | A
| ---------------- | ---
| HumHub version   | 1.7.1

**Comment by fae7em** [comment]
Actually, I'm not sure about the right scenario. I think, if a content is in a private container, it should be just displayed to the container members, and container is the owner of that content, not the user who create that content.

Currently, if a user create a content in a private space, even if this user cancel his/her membership in that space, he/she can access to that content in her profile, with access to delete the content, like post and its comments, add/edit/delete comments, see any new comment on that post. This user is not a member now, but can have activity in the space through her posts.

**Comment by buddh4** [comment]
Private space content should only be visible for space members, therefore there should not be public content in private spaces. We should force private content visibility for private spaces. Regarding my own content of a private space I'am not a member anymore, I think this is a very specific case and the user should still be able to see the content, otherwise I assume the visibility checks (which kind of are spread in different places unfortunately) will get too complex. Currently the rule is that users are always allowed to see his/her own content. In the future maybe we could add a way of change the ownership of such content. What do you think @luke-

**Comment by fae7em** [comment]
>  We should force private content visibility for private spaces in `beforeSave` (if its not the case already). 

No, it's not already, if space visibility change to private, its content visibility stay public, and as I mentioned in #4712, in the content `canView` function, it just checks if the content is public (and not its container) and return true.

**Comment by buddh4** [comment]
So we'd need to switch the visibility of all contents when changing the space visibility. Also we should render a warning that all content visibilities will be changed.

**Comment by luke-** [comment]
Yes, public content should not exist in a private Space. So we should switch existing public content (including its activities) to private when the Space setting is changed. (incl. a Warning of course). This process should ideally be executed as an ActiveJob. 
@fae7em  Would you like to start a new issue & PR for this?   

Because of the second issue, I also think that content should belong to the ContentContainer.  As an example, from a Wiki page that grows, the original creator should no longer have access if they are no longer part of the Space.  I think making the change everywhere consistently should not be a big problem.  However, we should then consider all places:Streams, Search and ContentActiveQuery here.

**Comment by buddh4** [comment]
A simple update query for all container content should not be too expensive, so maybe we could skip the ActiveJob?

**Comment by luke-** [comment]
@buddh4 Ok right, I thought for Activities we might need an extra query per content, but these are also directly assigned to the Space.
What about followers of the space or of public content?

**Comment by buddh4** [comment]
Hmm, yes perhaps an Activejob would be better, otherwise we can not handle cleaning content followers etc.

**Comment by fae7em** [comment]
I 

> Hmm, yes perhaps an Activejob would be better, otherwise we can not handle cleaning content followers etc.

I'm already working on followers in #4632 and I created a draft PR for it : #4683

**Comment by fae7em** [comment]
> @fae7em Would you like to start a new issue & PR for this?

Actually, I already create an issue for it in #4712 , because I thought it's a part of project logic that public content stays public, I create that issue to also check the container visibility in `canView`.
